test/core/client_channel/xds_override_host_lb_config_parser_test.cc

```diff
@@ -0,0 +1,166 @@
+  }
+};
+
+TEST_F(XdsOverrideHostConfigParsingTest, ValidConfig) {
+  const char* service_config_json =
+      "{\n"
+      "  \"loadBalancingConfig\":[{\n"
+      "    \"xds_override_host_experimental\":{\n"
+      "      \"childPolicy\":[\n"
+      "        {\"grpclb\":{}}\n"
+      "      ]\n"
+      "    }\n"
+      "  }]\n"
+      "}\n";
+  auto service_config =
+      ServiceConfigImpl::Create(ChannelArgs(), service_config_json);
+  ASSERT_TRUE(service_config.ok());
+  EXPECT_NE(*service_config, nullptr);
+}
+
+TEST_F(XdsOverrideHostConfigParsingTest, ReportsMissingChildPolicyField) {
+  const char* service_config_json =
+      "{\n"
+      "  \"loadBalancingConfig\":[{\n"
+      "    \"xds_override_host_experimental\":{\n"
+      "    }\n"
+      "  }]\n"
+      "}\n";
+  auto service_config =
+      ServiceConfigImpl::Create(ChannelArgs(), service_config_json);
+  ASSERT_FALSE(service_config.ok());
+  EXPECT_EQ(service_config.status(),
+            absl::InvalidArgumentError(
+                "errors validating service config: [field:loadBalancingConfig "
+                "error:errors validating xds_override_host LB policy config: "
+                "[field:childPolicy error:field not present]]"));
+}
+
+TEST_F(XdsOverrideHostConfigParsingTest, ReportsChildPolicyShouldBeArray) {
+  const char* service_config_json =
+      "{\n"
+      "  \"loadBalancingConfig\":[{\n"
+      "    \"xds_override_host_experimental\":{\n"
+      "      \"childPolicy\":{\n"
+      "        \"grpclb\":{},\n"
+      "      }\n"
+      "    }\n"
+      "  }]\n"
+      "}\n";
+  auto service_config =
+      ServiceConfigImpl::Create(ChannelArgs(), service_config_json);
+  ASSERT_FALSE(service_config.ok()) << service_config.status();
+  EXPECT_EQ(service_config.status(),
+            absl::InvalidArgumentError(
+                "errors validating service config: [field:loadBalancingConfig "
+                "error:errors validating xds_override_host LB policy config: "
+                "[field:childPolicy error:type should be array]]"));
+}
+
+TEST_F(XdsOverrideHostConfigParsingTest, ReportsEmptyChildPolicyArray) {
+  const char* service_config_json =
+      "{\n"
+      "  \"loadBalancingConfig\":[{\n"
+      "    \"xds_override_host_experimental\":{\n"
+      "      \"childPolicy\":[\n"
+      "      ]\n"
+      "    }\n"
+      "  }]\n"
+      "}\n";
+  auto service_config =
+      ServiceConfigImpl::Create(ChannelArgs(), service_config_json);
+  ASSERT_FALSE(service_config.ok()) << service_config.status();
+  EXPECT_EQ(service_config.status(),
+            absl::InvalidArgumentError(
+                "errors validating service config: [field:loadBalancingConfig "
+                "error:errors validating xds_override_host LB policy config: "
+                "[field:childPolicy error:No known policies in list: ]]"));
+}
+
+class XdsOverrideHostDisabledConfigParsingTest : public ::testing::Test {
+ public:
+  static void SetUpTestSuite() { grpc_init(); }
```

Comment:
If you take my suggestion above about moving `grpc_init()` and `grpc_shutdown()` to main, then I don't think we need these `SetUpTestSuite()` and `TearDownTestSuite()` methods.

Reply:
Yep. Removed.